src/s3/activation.ts

```diff
@@ -29,7 +35,53 @@ export async function activate(extensionContext: vscode.ExtensionContext): Promi
             await downloadFileAsCommand(node)
         }),
         vscode.commands.registerCommand('aws.s3.uploadFile', async (node: S3BucketNode | S3FolderNode) => {
-            await uploadFileCommand(node)
+            const fileLocation = await getFileToUpload()
+            if (!fileLocation) {
+                getLogger().info('UploadFile cancelled')
+                telemetry.recordS3UploadObject({ result: 'Cancelled' })
+                return
+            }
+            const key = node.path + path.basename(fileLocation.fsPath)
+            const bucket:S3.Bucket = { Name: node.bucket.name }
+            await uploadFileCommand(fileLocation, key, bucket, node.s3)
+        }),
+        vscode.commands.registerCommand('aws.s3.uploadFileToS3', async () => {
+            while (true) {
+                const editor = vscode.window.activeTextEditor
+                const document = editor?.document.uri
+                const file = await getFileToUpload(document)
+                if (file) {
+                    try{
+                        const regionCode = Workspace.vscode().getConfiguration(extensionSettingsPrefix).get<string>('s3.defaultRegion')?? "us-east-1"
```

Comment:
style: always use single-quotes. Check that prettier/husky are setup correctly on your machine.
```suggestion
                        const regionCode = Workspace.vscode().getConfiguration(extensionSettingsPrefix).get<string>('s3.defaultRegion')?? 'us-east-1'
```

Reply:
Probably will have to merge master into this branch to get the husky fix. Then run `npm i` again.